queueConversationControl/queueConversationControl.js

```diff
@@ -0,0 +1,187 @@
+            <td class="offered">?</td>
+            <td class="answered">?</td>cls
+        </tr>`)
+            .appendTo('#queueStatsTable > tbody:last-child');
+        miccSignalr.addQueueStateMonitor([queueData.id]);
+        miccSignalr.addQueueConversationsMonitor([queueData.id]);
+        populateQueueStats(queueData);
+        micc.getQueueConversations(queueData.id, function (queueConversations) {
+            addQueueConversationRows(queueConversations._embedded.items);
+        });
+    });
+}
+
+function addQueueConversationRows(conversations) {
+    for (var i = 0; i < conversations.length; i++) {
+        var conversation = conversations[i];
+        var matchingRows = $(`[id='${conversation.conversationId}']`);
+        if (matchingRows && matchingRows.length > 0) {
+            populateQueueConversationStats(conversation);
+        } else {
+            var conversationRow = $(`<tr id="${conversation.conversationId}" class="conversationRow">
+                ${getConversationActionButtons(conversation)}
+                <td class="${conversation.queueId}">${conversation.queueName}</td>            
+                <td class="conversationId">?</td>
+                <td class="mediaType">?</td>
+                <td class="caseId">?</td>
+                <td class="fromName">?</td>
+                <td class="fromAddress">?</td>
+                <td class="toName">?</td>
+                <td class="toAddress">?</td>
+                <td class="subject">?</td>
+                <td class="timeOfferedToSystem">${conversation.timeOfferedToSystem}</td>
+            </tr>`)
+                .appendTo('#queueConversationsTable > tbody:last-child');
+            populateQueueConversationStats(conversation);
+        }
+    }
+}
+
+function removeQueueConversationRows(conversations) {
+    for (var i = 0; i < conversations.length; i++) {
+        var matchingConversationsRows = $(`[id='${conversations[i]}']`);
+        if (matchingConversationsRows) {
+            matchingConversationsRows.each(function (i, conversationRow) {
+                var query = $(conversationRow);
+                query.remove();
+            });
+        }
+    }
+}
+
+function deleteQueue(button, id) {
+    miccSignalr.removeQueueStateMonitor([id]);
+    miccSignalr.removeQueueConversationsMonitor([id]);
+    // remove queue row
+    $(button).parent().parent().remove();
+    // remove queue conversations rows
+    var matchingQueueConversationsRows = $(`[class='${id}']`);
+    if (!matchingQueueConversationsRows) {
+        console.info('No conversations found for queue with id %s', id);
+    } else {
+        matchingQueueConversationsRows.each(function (i, conversationRow) {
+            $(conversationRow).parent().remove();
+        });
+    }
+}
+
+function getConversationActionButtons(conversation) {
+    var html = '<td>';
+    if (conversation) {
+        // Pick
+        html += `<button title="Pick" class="w3-btn w3-round-xxlarge w3-tiny w3-blue" 
```

Comment:
We can do better on the repeated code.  The html blocks are identical except for the title and function name.  Extract into a function to build the button html fragment.